docs/tutorials/creating-a-runtime-module.md

```diff
@@ -51,184 +73,286 @@ You should be able to successfully compile this template module with:
+you can update to a newer git commit of Substrate, recompile and test your
+module against the runtime, and then release your module.
 
-If you follow our recommended option 1, your dependency definitions will look like:
+In this case you change the Substrate module `Cargo.toml` to use
+`rev = <git commit hash>`, e.g.:
 
-```toml
-# This dependency is published on crates.io so just use a version
-[dependencies.parity-codec]
-default-features = false
-features = ['derive']
-version = '3.5'
+```TOML
+#--snip--
 
-# These dependencies specify a git branch
 [dependencies.support]
 default_features = false
 git = 'https://github.com/paritytech/substrate.git'
 package = 'srml-support'
-branch = 'v1.0'
+rev = '3dedd246c62255ba6f9b777ecba318dfc2078d85'
+
+#--snip--
+```
+
+This means that only nodes which use the exact same git commit hash for their
+projects can use your module.
+
+**Option 2**
+
+If you prefer to have more stability, you can develop your module against
+Substrate's known release tags (e.g.
+[`v1.0.0`](https://github.com/paritytech/substrate/tree/v1.0.0)). In this case
+you will need to update your node template's `Cargo.toml` to depend on
+Substrate's `tag = 'v1.0.0'`, e.g.:
 
-[dependencies.system]
+```TOML
+#--snip--
+
+[dependencies.support]
 default_features = false
 git = 'https://github.com/paritytech/substrate.git'
-package = 'srml-system'
-branch = 'v1.0'
-# Could also use a specific revision
-# rev = '<commit hash>'
+package = 'srml-support'
+tag = 'v1.0.0'
+
+#--snip--
 ```
 
 ### Your Module's Dev Dependencies
 
-The final section of the `Cargo.toml` file specifies the dev dependencies. These are the dependencies that are needed in your module's tests, but not necessary the actual module itself.
+The final section of the `Cargo.toml` file specifies the dev dependencies.
+These are the dependencies that are needed in your module's tests, but not
+necessary the actual module itself.
 
-
-```toml
+```TOML
+#--snip--
 
 [dev-dependencies.primitives]
 git = 'https://github.com/paritytech/substrate.git'
 package = 'substrate-primitives'
-branch = 'v1.0'
-
-[dev-dependencies.runtime-primitives]
-git = 'https://github.com/paritytech/substrate.git'
-package = 'sr-primitives'
-branch = 'v1.0'
-
-[dev-dependencies.runtime-io]
-git = 'https://github.com/paritytech/substrate.git'
-package = 'sr-io'
-branch = 'v1.0'
+rev = '3dedd246c62255ba6f9b777ecba318dfc2078d85'
 ```
 
-You can confirm for yourself that the Substrate module template's test passes with:
+You can confirm for yourself that the Substrate module template's test passes
+with:
 
 ```bash
 cargo test
 ```
 
-> Optional: You can remove these dev dependencies from your `Cargo.toml` file and confirm that the module still compiles with `cargo build`. However, you will notice that `cargo test` does not compile! If you do this experiment, remember to restore the dev dependencies afterward.
+> Optional: You can remove these dev dependencies from your `Cargo.toml` file
```

Comment:
IMO, remove this optional thing and just tell the user what will happen. Again, I think a tutorial should have a single well defined path.